snmp/tests/test_e2e_core_profiles/test_profile_bluecat_server.py

```diff
@@ -0,0 +1,97 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import pytest
+
+from datadog_checks.dev.utils import get_metadata_metrics
+
+from .. import common
+from ..test_e2e_core_metadata import assert_device_metadata
+from .utils import (
+    assert_common_metrics,
+    assert_extend_generic_host_resources,
+    assert_extend_generic_if,
+    create_e2e_core_test_config,
+    get_device_ip_from_config,
+)
+
+pytestmark = [pytest.mark.e2e, common.py3_plus_only, common.snmp_integration_only]
+
+
+def test_e2e_profile_bluecat_server(dd_agent_check):
+    config = create_e2e_core_test_config('bluecat-server')
+    aggregator = common.dd_agent_check_wrapper(dd_agent_check, config, rate=True)
+
+    ip_address = get_device_ip_from_config(config)
+    common_tags = [
+        'snmp_profile:bluecat-server',
+        'snmp_host:bluecat-server.device.name',
+        'device_namespace:default',
+        'snmp_device:' + ip_address,
+    ] + []
+
+    # --- TEST EXTENDED METRICS ---
+    # Examples:
+    assert_extend_generic_if(aggregator, common_tags)
+    assert_extend_generic_host_resources(aggregator, common_tags)
+
```

Comment:
can you also add a `assert_generic_ucd` as well ?

Reply:
Added a new test util!